server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/MinorCompactor.java

```diff
@@ -131,12 +134,21 @@ public CompactionStats call() {
           reportedProblem = true;
         } catch (RuntimeException | NoClassDefFoundError e) {
           // if this is coming from a user iterator, it is possible that the user could change the
-          // iterator config and that the
-          // minor compaction would succeed
-          log.warn("MinC failed ({}) to create {} retrying ...", e.getMessage(), outputFileName, e);
+          // iterator config and that the minor compaction would succeed
+          // If the minor compaction stalls for too long during recovery, it can interfere with
+          // other tables loading
+          // Throw exception if this happens so assignments can be rescheduled.
           ProblemReports.getInstance(tabletServer.getContext()).report(
               new ProblemReport(getExtent().tableId(), ProblemType.FILE_WRITE, outputFileName, e));
+          if (retryCounter >= 4 && mincReason.equals(MinorCompactionReason.RECOVERY)) {
```

Comment:
Should the same thing be done in the preceding catch block?  Also before this change, it seems like this catch block and the preceding catch block did the exact same thing.  Curios if you see any diffs @Manno15 w/ the previous catch block.  If they were the same and we want to do this check in both, then maybe the two catch blocks could be collapsed into one.

Reply:
ok I see a diff now, the 2nd catch block passes an exception to log.warn().  Still wondering if the prev catch block should do the same check.

Reply:
I am not sure it is necessary. The issue stems from a misconfigured user iterator which I believe will only ever trigger the second catch block. I will look into it and see what would throw an `IOException` or `UnsatisfiedLinkError` and see if it relates to recovery or not.

Reply:
> I am not sure it is necessary. The issue stems from a misconfigured user iterator which I believe will only ever trigger the second catch block.

Given the uncertainty about adding it to the other catch block, and that the observed problem was addressed, I think this can be merged for now and the other block could be modified later if needed.